app/src/main/java/com/github/sdp/ratemyepfl/fragment/navigation/HomeFragment.kt

```diff
@@ -0,0 +1,78 @@
+package com.github.sdp.ratemyepfl.fragment.navigation
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import android.widget.TextView
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import com.github.sdp.ratemyepfl.R
+import com.github.sdp.ratemyepfl.activity.SplashScreen
+import com.github.sdp.ratemyepfl.viewmodel.HomeViewModel
+import com.google.android.material.button.MaterialButton
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class HomeFragment : Fragment(R.layout.fragment_home) {
+
+    private lateinit var helloUser: TextView
+    private lateinit var connectionButton: MaterialButton
+
+    private val viewModel: HomeViewModel by viewModels()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        helloUser = view.findViewById(R.id.homePageHelloUserText)
+        connectionButton = view.findViewById(R.id.homePageConnectionButton)
+        setupConnectionButton()
+        setupUser()
+    }
+
+
+    private fun setupUser() {
+        viewModel.isUserLoggedIn.observe(viewLifecycleOwner) { loggedIn ->
+            val icon: Int =
+                if (loggedIn) R.drawable.ic_logout_black_24dp else R.drawable.ic_login_black_24dp
+            connectionButton.setIconResource(icon)
+
+            connectionButton.text = if (loggedIn) LOGOUT else LOGIN
+
+            val userName = viewModel.user.getEmail() ?: VISITOR_NAME
+            helloUser.text = getString(R.string.home_page_hello_user_text, userName)
+        }
+    }
+
+    private fun checkUser() {
+        viewModel.updateLoginStatus()
+    }
+
+    private fun setupConnectionButton() {
+        connectionButton.setOnClickListener {
+            if (context != null) {
+                if (viewModel.isUserLoggedIn()) {
+                    viewModel.signOut(requireContext())
```

Comment:
Yes, of course, we can. It's true that for now, the user had to start with the login. I think that for a better user experience, the user should be able to go through the app in any case (since login is not mandatory). I find that logout and then click on continue as a visitor is quite heavy. But tell me if you think I need to put it back.

Reply:
For now we can leave it like this, we can discuss this with the whole team.